src/core_plugins/vega/public/data_model/es_query_parser.js

```diff
@@ -81,13 +101,25 @@ export class EsQueryParser {
       }
       result += '}';
 
-      this._onWarning(
-        `Legacy "url": {"${LEGACY_CONTEXT}": ${JSON.stringify(legacyContext)}} should change to ${result}`);
+      this._onWarning(i18n.translate('vega.esQueryParser.legacyUrlShouldChangeToWarningMessage', {
+        defaultMessage: 'Legacy "url": {legacyUrl} should change to {result}',
```

Comment:
`url` is a Vega property in the spec. So we might want to pull this out into a param, to be aligned with the rest of param names.